### ISSUE TITLE
🐛 Bug Report: Running yarn dev starts only frontend customized plugin and not its backend in port 7007

### ISSUE THREAD

**Issue by eilonash92** [backstage/backstage#24696]
### 📜 Description

I develop a frontend plugin and a backend plugin.
When I start each plugin by itself using yarn start, I'm able to run both backend and frontend plugins.

But from the main page using yarn dev, only the frontend plugin is accessible, but backend plugin in port 7007 is not.

### 👍 Expected behavior

I will be able to run my plugin from the home page and the frontend plugin will be able to reach the backend plugin data

### 👎 Actual Behavior with Screenshots

![image](https://github.com/backstage/backstage/assets/29045521/118c33ee-45f2-48b2-8312-2bd9c489e2cc)

### 👟 Reproduction steps

Going into backstage directory and running yarn dev doesn't start my backend plugin, only the core plugins and my frontend plugin.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by Rugvip** [comment]
Have you added the plugin to `packages/backend`? When you `yarn start` in the plugin folders you'll use a local setup for the plugin, but `yarn dev` in the repo root will instead run `packages/app` and `packages/backend`.

**Comment by eilonash92** [comment]
If its a manual procedure, shouldn't we want to add it automatically to packages/backend when we create backstage plugin?
Since we do it automatically to frontend plugins in packages/app

**Comment by eilonash92** [comment]
@Rugvip adding it to packages/backend solved the issue.

**Comment by Rugvip** [comment]
@eilonash92 yep it is right now, we'll likely adjust one of the two in the future so that they both match. Either not adding the frontend plugin to the app, or adding the backend plugin to the backend.
